pkg/commands/compute/init.go

```diff
@@ -196,25 +215,130 @@ func (c *InitCommand) Exec(in io.Reader, out io.Writer) (err error) {
-	// This can happen in two ways:
+	// There are three situations in which we might fetch something
+	// here. We might fetch a template if:
 	//
-	// 1. --from flag is set
+	// 1. --from flag is set to a template repository, or
 	// 2. user selects starter kit when prompted
 	//
+	// Or we fetch an existing, deployed package if
+	//
+	// 3. --from flag is set to a serviceID
+	//
 	// We don't fetch if the user has indicated their language of choice is
 	// "other" because this means they intend on handling the compilation of code
 	// that isn't natively supported by the platform.
 	if c.cloneFrom != "" {
-		err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
-		if err != nil {
-			c.Globals.ErrLog.AddWithContext(err, map[string]any{
-				"From":      from,
-				"Branch":    branch,
-				"Tag":       tag,
-				"Directory": c.dir,
+		if !isExistingService {
+			err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
+			if err != nil {
+				c.Globals.ErrLog.AddWithContext(err, map[string]any{
+					"From":      from,
+					"Branch":    branch,
+					"Tag":       tag,
+					"Directory": c.dir,
+				})
+				return err
+			}
+		} else {
+			// profileName, _, _ := c.Globals.Profile()
+
+			var (
+				serviceDetails *fastly.ServiceDetail
+				pack           *fastly.Package
+			)
+			err = spinner.Process("Fetching service details", func(_ *text.SpinnerWrapper) error {
+				serviceDetails, err = c.Globals.APIClient.GetServiceDetails(&fastly.GetServiceInput{
+					ServiceID: c.cloneFrom,
+				})
+				if err != nil {
+					c.Globals.ErrLog.AddWithContext(err, map[string]any{
+						"From":      c.cloneFrom,
+						"Directory": c.dir,
+					})
+					if hErr, ok := err.(*fastly.HTTPError); ok && hErr.IsNotFound() {
+						return fmt.Errorf("the service %s could not be found", c.cloneFrom)
+					}
+					return err
+				}
+
+				if serviceDetails.Type == nil || *serviceDetails.Type != "wasm" {
+					return fmt.Errorf("service %s is not a Compute service (type is %s)", c.cloneFrom, fastly.ToValue(serviceDetails.Type))
+				}
+
+				var serviceVersion int
+				if serviceDetails.ActiveVersion != nil {
+					serviceVersion = fastly.ToValue(serviceDetails.ActiveVersion.Number)
+					pack, err = c.Globals.APIClient.GetPackage(&fastly.GetPackageInput{
+						ServiceID:      c.cloneFrom,
+						ServiceVersion: serviceVersion,
+					})
+					if err != nil {
+						return err
+					}
+				} else {
+					for _, v := range serviceDetails.Versions {
+						if v.Number != nil && *v.Number > serviceVersion {
+							serviceVersion = *v.Number
+
+							pack, err = c.Globals.APIClient.GetPackage(&fastly.GetPackageInput{
+								ServiceID:      c.cloneFrom,
+								ServiceVersion: serviceVersion,
+							})
+							if err != nil {
+								if hErr, ok := err.(*fastly.HTTPError); ok {
+									if hErr.IsNotFound() {
+										continue
+									}
+								}
+								return err
+							}
+						}
+					}
```

Comment:
We should add a `break` directive so we stop the loop once a version is found.

Also, do we need the `*v.Number > serviceVersion` check? I'm guessing if there is any length to `serviceDetails.Version` then that would suggest a valid service version is assigned to `v`. 

We could also use `fastly.ToValue()` instead of the nil check.

```go
for _, v := range serviceDetails.Versions {
	serviceVersion = fastly.ToValue(v.Number)
	
	// ...rest of code...
}
```

Reply:
Ah I might have extrapolated something from the [changelog](https://github.com/fastly/go-fastly/blob/v9.3.1/CHANGELOG.md#v900-2024-02-05) of `go-fastly` here: if the Versions are sorted, then yes we can stop once we find a good version. Otherwise, this is looking for the highest version number with a package attached.

Reply:
Looks like they're sorted by number ascending—I'll revise this to start at the end and break early